### PR TITLE
IN-1075 Stop migration if Sirius has non-CP1 finance data

### PR DESCRIPTION
**Pull request by devkokov**
## Purpose

IN-1075

## Approach

Extend the deletion script to find any finance records linked to non-CP1 clients and insert them into the deletions schema, however there are no DELETE statements to delete these records automatically.

After the deletion script has run, check that there's no finance records in the deletion schema. If there are any, output the SQL query that can be run manually in prod to delete them, and stop migration.

Stop truncating the events table in QA so we know how long migration will actually take in prod, and build confidence that the events deletion script works.

Run the deletion script in all envs, including production 😬 

Example output:
```
2022-01-13T14:24:11UTC app.py             INFO    : Checking deletion table counts
2022-01-13T14:24:11UTC app.py             INFO    : Found 0 records in deletions.deletions_finance_invoice. Expected 0.
2022-01-13T14:24:11UTC app.py             INFO    : Found 0 records in deletions.deletions_finance_ledger. Expected 0.
2022-01-13T14:24:11UTC app.py             INFO    : Found 0 records in deletions.deletions_finance_ledger_allocation. Expected 0.
2022-01-13T14:24:11UTC app.py             INFO    : Found 0 records in deletions.deletions_finance_remission_exemption. Expected 0.
2022-01-13T14:24:11UTC app.py             INFO    : Found 7 records in deletions.deletions_finance_order. Expected 0.
2022-01-13T14:24:11UTC app.py             INFO    : To delete these records manually, run:
2022-01-13T14:24:11UTC app.py             INFO    : DELETE FROM finance_order WHERE id IN (1,2,3,4,5,6,7);
2022-01-13T14:24:11UTC app.py             ERROR   : Stopping migration due to unexpected deletion counts. Check log output above.
ERROR: 1
```

## Checklist

* [ ] I have performed a self-review of my own code
* [ ] I have done an adhoc run against preprod (only needed for high complexity PRs)
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: migration_steps/initialise_environments/delete_existing_data/app.py [hunk 1]
```diff
@@ -70,13 +91,28 @@ def main():
...
+        log.info(f"{response.rowcount} rows updated\n")
+
+    log.info("Checking deletion table counts")
+    stop_migration = False
+    for table in sirius_deletion_check_tables:
+        statement = f"SELECT id FROM deletions.deletions_{table};"
+        response = target_db_engine.execute(statement)
+        row_count = response.rowcount
+        log.info(f"Found {row_count} records in deletions.deletions_{table}. Expected 0.")
+        if row_count == 0:
```
Comment (townxelliot) [inline migration_steps/initialise_environments/delete_existing_data/app.py:106]: My personal preference would be to do something like:
```
if row_count > 0:
  ...stop the migration here...
```
I find the `continue` syntax a bit misleading in this context.

**Review by townxelliot — COMMENTED** [review]
lgtm, but would feel more comfortable if @jamesrwarren checked the deletion logic, as I'm not so familiar with it

File: migration_steps/initialise_environments/delete_existing_data/app.py (hunk 1 above)
Comment (devkokov) [reply migration_steps/initialise_environments/delete_existing_data/app.py:106]: Agreed, it's not needed here. I've got a habit of exiting ifs early to keep a flatter code structure - some bad previous experience of massive if statements nested several levels deep :D

File: migration_steps/initialise_environments/initialise_environments.sh [hunk 2]
```diff
@@ -12,18 +12,7 @@ done
...
-  || [ "${ENVIRONMENT}" == "preproduction" ] \
-  || [ "${ENVIRONMENT}" == "preqa" ] \
-  || [ "${ENVIRONMENT}" == "qa" ]
-  then
-  python3 "${DIR}/delete_existing_data/app.py"
-else
-  echo "delete_existing_data should not run on ${ENVIRONMENT}"
-fi
-
+python3 "${DIR}/delete_existing_data/app.py"
```
Comment (jamesrwarren) [inline migration_steps/initialise_environments/initialise_environments.sh:15]: This makes me nervous

**Review by jamesrwarren — APPROVED** [review]
Going to have to work out how we roll out the env to prod without actually doing anything on prod but this PR looks good.

File: migration_steps/initialise_environments/initialise_environments.sh (hunk 2 above)
Comment (devkokov) [reply migration_steps/initialise_environments/initialise_environments.sh:15]: I'd be nervous too if it was my script! 😄  On a serious note - we've had several people look at it and sign off on it + there's the assurance from Jack's count checks and UAT. Should be fine

**Comment by devkokov** [comment]
> Going to have to work out how we roll out the env to prod without actually doing anything on prod but this PR looks good.

Do we? The preprod run should capture any issues beforehand

**Pull request merged**